spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/DefaultSftpSessionFactory.java

```diff
@@ -320,7 +320,7 @@ private ClientSession initClientSession() throws IOException {
 		return clientSession;
 	}
 
-	private void initClient() throws IOException {
+	private synchronized void initClient() throws IOException {
 		if (this.initialized.compareAndSet(false, true)) {
```

Comment:
`this.initialized` should not be set at the beginning of the method.
If initialisation fails, the variable will be set to true for un this.initialized client

Reply:
Yeah... Makes sense.
OK closing my PR.
Feel free to copy my unit test into your PR then we will review it and merge properly.